### PR TITLE
[GSoC 2020] change output in standalone mode to standard mode (hour/minute/seconds)

### LINKED ISSUE #1162: Improve report function in C++ standalone mode
[Note that this issue is "reserved" for prospective GSoC students and probably there will be multiple pull requests open at the same time. No pull request will be merged before the end of the application period.]

When you use `report='text'` with a run (see the [documentation on progress reports](https://brian2.readthedocs.io/en/stable/user/running.html#progress-reporting) and [custom reports](https://brian2.readthedocs.io/en/stable/advanced/scheduling.html#id1)), you get a standard progress report that tells you how much of the simulation has been already done, how much time was spent, and an estimate of the remaining time. In runtime mode, the simulated and estimated time are formatted in hours/minutes/seconds for readability:
```
18.2041 s (1%) simulated in 1m 0s, estimated 53m 56s remaining.
```
In standalone mode, the display simply shows the times in seconds:
```
37.4416 s (3%) simulated in 70 s, estimated 1800 s remaining.
```

**It would be nice to be consistent here and show the time nicely formatted in standalone mode as well.**

*Bonus* improvements:

* the [documentation](https://brian2.readthedocs.io/en/stable/advanced/scheduling.html#custom-progress-reporting) currently does not have any interesting example for a custom progress report function in C++ standalone. It would be nice to add one, e.g. one that writes the progress to a file or one that displays a simple text-based progress bar on the terminal.
* we might want to implement some simple(!) approach to improve the estimates of the remaining time (this would also apply to runtime mode). See e.g. https://stackoverflow.com/questions/933242/smart-progress-bar-eta-computation for some pointers.

### PR DESCRIPTION
**Pull request by rahuliitg**
fixes #1162 
for Gsoc 2020

### CODE REVIEW

**Comment by mstimberg** [comment]
@rahuliitg Thanks for the pull request, please let me know with a short comment whether you are still working on it or whether you consider it finished from your side. If it is basically finished, I'd start reviewing it.

**Comment by rahuliitg** [comment]
@mstimberg, I've done from my side

**Comment by rahuliitg** [comment]
@mstimberg, I have earlier submitted the patch using std::string but it failed the initial build test cases stating that - "to_string is not a member of std"

**Comment by mstimberg** [comment]
> @mstimberg, I have earlier submitted the patch using std::string but it failed the initial build test cases stating that - "to_string is not a member of std"

This was probably before you moved the `#include` to `main.cpp`?

**Comment by rahuliitg** [comment]
ok I'll do try with std::to_string

**Comment by rahuliitg** [comment]
Screenshot 2020-03-12 at 10 21 38 PM
got this error. this is a known bug under MinGW (https://stackoverflow.com/questions/12975341/to-string-is-not-a-member-of-std-says-g-mingw)

**Comment by mstimberg** [comment]
> this is a known bug under MinGW (https://stackoverflow.com/questions/12975341/to-string-is-not-a-member-of-std-says-g-mingw)

We are not using MinGW, though...

**Comment by mstimberg** [comment]
It works on my machine, but it might be a g++ version issue. Could you try adding `-std=c++11` to the standard options we pass to g++? The default options are defined here:
https://github.com/brian-team/brian2/blob/068196850fd21b615edf668bb6872d37b638e70f/brian2/codegen/cpp_prefs.py#L113

**Comment by rahuliitg** [comment]
> It works on my machine, but it might be a g++ version issue. Could you try adding `-std=c++11` to the standard options we pass to g++? The default options are defined here:
> https://github.com/brian-team/brian2/blob/068196850fd21b615edf668bb6872d37b638e70f/brian2/codegen/cpp_prefs.py#L113

ok trying this approach.

File: brian2/devices/cpp_standalone/templates/main.cpp [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 #include <iostream>
 #include <fstream>
+#include <sstream>
+#include <string>
```
Comment (mstimberg) [inline brian2/devices/cpp_standalone/templates/main.cpp]: I think `sstream` was only needed for your `to_str` function, right?
```suggestion
#include <string>
```

File: brian2/devices/cpp_standalone/templates/network.cpp [hunk 2]
```diff
@@ -5,6 +5,9 @@
 #include<iostream>
 #include <ctime>
 #include<utility>
+#include<string>
+#include <sstream>
+
```
Comment (mstimberg) [inline brian2/devices/cpp_standalone/templates/network.cpp]: I'm not 100% sure, but I think it is enough to have these includes in `main.cpp`
```suggestion
```

Comment (rahuliitg) [reply brian2/devices/cpp_standalone/templates/network.cpp]: yes, I missed removing # from network.cpp file

File: brian2/devices/cpp_standalone/templates/main.cpp (hunk 1 above)
Comment (rahuliitg) [reply brian2/devices/cpp_standalone/templates/main.cpp]: yes

**Comment by mstimberg** [comment]
Apologies for the long delay in reviewing. I did not want to give further comments on PRs made as part of GSoC applications after the application deadline, to not give participants the false impression that this work was still taken into account for the application. I am going to close all "GSoC pull request" now, but of course we'd be more than happy to see you as a contributor in the future.

All that said, changing the displayed times in standalone mode was a real issue, and I am going to merge this PR since it fixes the issue and is the most concise of all contributions. Thanks again!

**Pull request merged**